staging/src/k8s.io/cloud-provider/controllers/node/node_controller_test.go

```diff
@@ -708,6 +1226,87 @@ func Test_AddCloudNode(t *testing.T) {
 	}
 }
 
+// test AddCloudNode with instanceV2, same test case with TestGCECondition.
+func TestGCEConditionV2(t *testing.T) {
+	existingNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "node0",
+			CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
+		},
+		Status: v1.NodeStatus{
+			Conditions: []v1.NodeCondition{
+				{
+					Type:               v1.NodeReady,
+					Status:             v1.ConditionUnknown,
+					LastHeartbeatTime:  metav1.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
+					LastTransitionTime: metav1.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
+				},
+			},
+		},
+		Spec: v1.NodeSpec{
+			Taints: []v1.Taint{
+				{
+					Key:    cloudproviderapi.TaintExternalCloudProvider,
+					Value:  "true",
+					Effect: v1.TaintEffectNoSchedule,
+				},
+			},
+		},
+	}
+
+	fakeCloud := &fakecloud.Cloud{
+		EnableInstancesV2: false,
```

Comment:
It would be `EnableInstancesV2: true`, I will address it later(together with other feedbacks).